### PR TITLE
Support non-privileged packaged apps (fixes #116)

### LINKED ISSUE #116: Support in-app payments for packaged app type:web
For one of our partner launches, we need to support in-app payments for a packaged app of type:web which is not currently possible because no origin can be defined.

### PR DESCRIPTION
**Pull request by kumar303**
This depends on a zamboni patch but it won't
break anything until then. Web packages were always
broken, they just never returned product results.

### CODE REVIEW

File: lib/fxpay/utils.js [hunk 1]
```diff
@@ -34,11 +34,39 @@ define([
...
-      // If that's a requirement, the caller should check for nulls.
+      // Check the manifest-declared origin (not the auto-generated one).
+      var hasDeclaredOrigin = !!settingsObj.appSelf.manifest.origin;
+
+      if (settingsObj.appSelf.manifest.type === 'web' || !hasDeclaredOrigin) {
+        // This package does not have a reliable origin so
+        // here we look for its marketplace hosted package URL
+        // to derive a marketplace-specific origin.
+        var pat = new RegExp(
+          '^https?://(marketplace|mp\\.dev).*/app/([^/]+)/manifest\\.webapp$'
```
Comment (muffinresearch) [inline lib/fxpay/utils.js]: Could make the first group a non-capturing group.

Comment (muffinresearch) [inline lib/fxpay/utils.js]: Fwiw .dev is a real tld - albeit apparently only for private usage.

**Comment by muffinresearch** [comment]
lgtm r+wc

File: Gruntfile.js [hunk 2]
```diff
@@ -104,18 +104,36 @@ module.exports = function(grunt) {
         dest: 'build/app/',
         expand: true,
       },
-      'example-shared': {
+      'example-packaged-web': {
+        cwd: 'example/packaged-web/',
+        src: '*',
```
Comment (muffinresearch) [inline Gruntfile.js]: Are all the uses of '*' always going to be one level deep?

**Comment by kumar303** [comment]
zamboni support: https://github.com/mozilla/zamboni/pull/3187

**Pull request merged**